toolchain/check/handle_pattern_binding.cpp

```diff
@@ -77,29 +77,31 @@ auto HandlePatternBinding(Context& context, Parse::Node parse_node) -> bool {
           })) {
         cast_type_id = SemIR::TypeId::Error;
       }
-      SemIR::NodeId value_id = SemIR::NodeId::Invalid;
+      SemIR::InstId value_id = SemIR::InstId::Invalid;
       SemIR::TypeId value_type_id = cast_type_id;
       if (enclosing_class_decl) {
         auto& class_info =
             context.classes().Get(enclosing_class_decl->class_id);
-        auto field_type_node_id = context.AddNode(SemIR::UnboundFieldType{
-            parse_node, context.GetBuiltinType(SemIR::BuiltinKind::TypeType),
-            class_info.self_type_id, cast_type_id});
-        value_type_id = context.CanonicalizeType(field_type_node_id);
-        value_id = context.AddNode(
+        SemIR::InstId field_type_inst_id =
```

Comment:
Preserve `auto` for now:
```suggestion
        auto field_type_inst_id =
```